### PR TITLE
pythonPackages.sh: 1.11 -> 1.12.14

### PR DESCRIPTION
**Pull request by sveitser**
###### Motivation for this change
Package update. 

###### Things done

- "Modernize" to the best of my knowledge. 
- Enable unit tests.
- Add MIT license.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS (python2 and 3)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6729,18 +6729,25 @@ in {
...
-    src = pkgs.fetchurl {
-      url = "mirror://pypi/s/sh/${name}.tar.gz";
-      sha256 = "590fb9b84abf8b1f560df92d73d87965f1e85c6b8330f8a5f6b336b36f0559a4";
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "1z2hx357xp3v4cv44xmqp7lli3frndqpyfmpbxf7n76h7s1zaaxm";
     };
 
-    doCheck = false;
+    buildInputs = with self; [ coverage ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: checkInputs

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -6729,18 +6729,25 @@ in {
...
+
+    # A test needs the HOME directory to be different from $TMPDIR.
+    preCheck = ''
+    HOME=$(mktemp -d)
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: indentation

**Review by FRidh — CHANGES_REQUESTED** [review]
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

**Comment by sveitser** [comment]
@FRidh thanks. I made the changes.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.sh python3.pkgs.sh

**Pull request merged**